# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/wasm/jsapi/function/type.tentative.any.js
# Commit: 3b62bd57747f
# Full Hash: 3b62bd57747f21c800a93da325d999fa5efb3850
# Author: Jessica Tallon <jtallon@igalia.com>
# Date: 2021-09-03 03:51:58
# Regressor Bug: 1715275
# File Overlap Count: 1
# Description:
#   Bug 1715275 - Add support for a JS WebAssembly.Function object r=rhunt
#   
#   This introduces a new Constructor WebAssembly.Function which is part of
#   the WASM JS-API type reflection work. This allows normal JS functions to
#   be wrapped with a type and used with WebAssembly objects (e.g. Tables).
# ==============================================================================

diff -r f6f4514d09bf -r 3b62bd57747f testing/web-platform/tests/wasm/jsapi/function/type.tentative.any.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/wasm/jsapi/function/type.tentative.any.js	Thu Sep 02 15:58:18 2021 +0000
@@ -0,0 +1,28 @@
+// META: global=window,dedicatedworker,jsshell
+// META: script=/wasm/jsapi/assertions.js
+
+function addNumbers(x, y, z) {
+    return x+y+z;
+}
+
+function doNothing() {}
+
+function assert_function(functype, func) {
+    var wasmFunc = new WebAssembly.Function(functype, func);
+    assert_equals(functype.parameters.length, wasmFunc.type().parameters.length);
+    for(let i = 0; i < functype.parameters.length; i++) {
+        assert_equals(functype.parameters[i], wasmFunc.type().parameters[i]);
+    }
+    assert_equals(functype.results.length, wasmFunc.type().results.length);
+    for(let i = 0; i < functype.results.length; i++) {
+        assert_equals(functype.results[i], wasmFunc.type().results[i]);
+    }
+}
+
+test(() => {
+    assert_function({results: [], parameters: []}, doNothing);
+}, "Check empty results and parameters")
+
+test(() => {
+    assert_function({results: ["f64"], parameters: ["i32", "i64", "f32"]}, addNumbers)
+}, "Check all types")
